zookeeper/dataset.py

```diff
@@ -0,0 +1,138 @@
+from typing import Optional
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from zookeeper import Component
+
+
+def base_splits(split):
+    """
+    Splits can be merged, e.g. `tfds.Split.TRAIN + tfds.Split.Validation`. For
+    such composite splits, find and return a list of 'base splits'.
+    """
+
+    if isinstance(split, tfds.core.splits._SplitMerged):
+        return base_splits(split._split1) + base_splits(split._split2)
```

Comment:
Should we instead support the new [string based Split API of TFDS](https://github.com/tensorflow/datasets/blob/master/docs/splits.md#s3-slicing-api)?

Reply:
I had not seen this before. To be quite honest I think the old API is far nicer; encoding the same thing in strings is ugly. I will investigate though.

Reply:
I also don't know if all datasets implement this API, so might be worth taking a look at it.